### PR TITLE
KYLIN-3574 Close underlying resource of lookup table

### PR DESCRIPTION
**Pull request by hit-lacus**
HTable should be closed.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: core-cube/src/main/java/org/apache/kylin/cube/cli/DictionaryGeneratorCLI.java [hunk 1]
```diff
@@ -98,7 +100,10 @@ private static void processSegment(KylinConfig config, CubeSegment cubeSeg, Stri
             logger.info("Checking snapshot of " + lookup);
             try {
                 JoinDesc join = cubeSeg.getModel().getJoinsTree().getJoinByPKSide(lookup);
-                cubeMgr.getLookupTable(cubeSeg, join);
+                ILookupTable table = cubeMgr.getLookupTable(cubeSeg, join);
+                if(table != null){
```
[the remark this answers is not in the document]
Comment (shaofengshi) [reply core-cube/src/main/java/org/apache/kylin/cube/cli/DictionaryGeneratorCLI.java]: Seems its code format is invalid; Please check https://kylin.apache.org/development/dev_env.html to enable code formatter.

[the remark this answers is not in the document]
Comment (hit-lacus) [reply core-cube/src/main/java/org/apache/kylin/cube/cli/DictionaryGeneratorCLI.java]: Sorry, I will fix it.

**Pull request merged**